test-utils/runtime/src/lib.rs

```diff
@@ -284,349 +156,153 @@ pub type Block = sp_runtime::generic::Block<Header, Extrinsic>;
+		fn test_sr25519_crypto() -> (sr25519::AppSignature, sr25519::AppPublic);
+		/// Test that `ecdsa` crypto works in the runtime.
+		///
+		/// Returns the signature generated for the message `ecdsa`.
+		fn test_ecdsa_crypto() -> (ecdsa::AppSignature, ecdsa::AppPublic);
+		/// Run various tests against storage.
+		fn test_storage();
+		/// Check a witness.
+		fn test_witness(proof: StorageProof, root: crate::Hash);
+		/// Test that ensures that we can call a function that takes multiple
+		/// arguments.
+		fn test_multiple_arguments(data: Vec<u8>, other: Vec<u8>, num: u32);
+		/// Traces log "Hey I'm runtime."
+		fn do_trace_log();
+		/// Verify the given signature, public & message bundle.
+		fn verify_ed25519(sig: ed25519::Signature, public: ed25519::Public, message: Vec<u8>) -> bool;
+	}
+}
+
+pub type Executive = frame_executive::Executive<
+	Runtime,
+	Block,
+	frame_system::ChainContext<Runtime>,
+	Runtime,
+	AllPalletsWithSystem,
+>;
+
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct SignedExtraDummy;
+
+impl sp_runtime::traits::Printable for SignedExtraDummy {
+	fn print(&self) {
+		"SignedExtraDummy".print()
+	}
+}
+
+impl sp_runtime::traits::Dispatchable for SignedExtraDummy {
+	type RuntimeOrigin = SignedExtraDummy;
+	type Config = SignedExtraDummy;
+	type Info = SignedExtraDummy;
+	type PostInfo = SignedExtraDummy;
+	fn dispatch(
 		self,
-		_f: impl FnOnce(Self::PalletsOrigin) -> Result<R, Self::PalletsOrigin>,
-	) -> Result<R, Self> {
-		unimplemented!("Not required in tests!")
-	}
-
-	fn none() -> Self {
-		unimplemented!("Not required in tests!")
-	}
-	fn root() -> Self {
-		unimplemented!("Not required in tests!")
-	}
-	fn signed(_by: Self::AccountId) -> Self {
-		unimplemented!("Not required in tests!")
-	}
-	fn as_signed(self) -> Option<Self::AccountId> {
-		unimplemented!("Not required in tests!")
-	}
-	fn as_system_ref(&self) -> Option<&RawOrigin<Self::AccountId>> {
-		unimplemented!("Not required in tests!")
-	}
-}
-
-#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo)]
-pub struct RuntimeEvent;
-
-impl From<frame_system::Event<Runtime>> for RuntimeEvent {
-	fn from(_evt: frame_system::Event<Runtime>) -> Self {
-		unimplemented!("Not required in tests!")
+		_origin: Self::RuntimeOrigin,
+	) -> sp_runtime::DispatchResultWithInfo<Self::PostInfo> {
+		panic!("This implementation should not be used for actual dispatch.");
 	}
 }
 
-impl frame_support::traits::PalletInfo for Runtime {
-	fn index<P: 'static>() -> Option<usize> {
-		let type_id = sp_std::any::TypeId::of::<P>();
-		if type_id == sp_std::any::TypeId::of::<system::Pallet<Runtime>>() {
-			return Some(0)
-		}
-		if type_id == sp_std::any::TypeId::of::<pallet_timestamp::Pallet<Runtime>>() {
-			return Some(1)
-		}
-		if type_id == sp_std::any::TypeId::of::<pallet_babe::Pallet<Runtime>>() {
-			return Some(2)
-		}
+impl sp_runtime::traits::SignedExtension for SignedExtraDummy {
```

Comment:
Not sure we need this?

Reply:
Not in this form. But we need it for implementing the control over `priority` and `propagate` which is required by some tests.

https://github.com/michalkucharczyk/substrate/blob/9aadd2ad501e9f09b1e96586082e4b4090fe8f60/test-utils/runtime/src/substrate_test_pallet.rs#L238-L247